ksql-engine/src/main/java/io/confluent/ksql/util/ExecutorWithRetries.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.common.errors.RetriableException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+
+public class ExecutorWithRetries {
+
+  private static final int NUM_RETRIES = 5;
+  private static final int RETRY_BACKOFF_MS = 500;
+  private static final Logger log = LoggerFactory.getLogger(ExecutorWithRetries.class);
+
+  public static <T> T execute(final Supplier<? extends Future<T>> supplier) throws Exception {
```

Comment:
Thanks for the updates @uurl .. This is indeed an improvement, since the retry logic is now in one place rather than two. However, the core loop is still similar and the logic is duplicated in both methods. It may be worth refactoring further so that there is one copy of the loop and the only difference is in either calling `supplier.get().get()` or `callable.call()`.